CMakeLists.txt

```diff
@@ -71,10 +71,11 @@ set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake ${CMAKE_MODULE_PATH}) #
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}) # Put binaries into root of build tree
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib) # Put libraries into their own dir
 
-set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -march=native")
-set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -march=native")
-set(CMAKE_C_FLAGS_RELWITHDEBINFO "${CMAKE_C_FLAGS_RELWITHDEBINFO} -march=native")
-set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO} -march=native")
+set(FLAGS_RELEASE "-march=native -fopenmp-simd")
```

Comment:
I'd prefer it if all our compile options were documented, otherwise it'll be hard to reason about why they were added (I, e.g., cannot tell what either of these do (okay, I could look that up) and why we'd need them)

Reply:
I'm all for documentation, but I'm not sure what to write here. What would you add to "Enable handling of OpenMP’s SIMD directives with #pragma omp in C/C++ [...]"?

Reply:
yes, I think something like that would help future me decide whether or not a flag is still needed or not.

Reply:
But that is literally the explanation of that flag. I am all in favor of adding comments that explain why we do something, how we do something, or that explain the big picture. I wouldn't start explaining external functions - once we start that, where do we draw the line?

Reply:
mmh, I'd probably feel more comfortable with that doc in because build scripts are so obscure (as opposed to external functions whose IO I can still monitor in a debugger).

Maybe -fopenmp-simd is simple, though for me personally it would help if a comment pointed to "we do this to auto vectorize some loops".

Different example: Googling "-march=native", e.g., yields no result. Googling "clang arch native" gives a bit of SO, but no real explanation. After diving into the man page (which is keeping things vague, too) I **assume** it's about letting the compiler do optimizations it wouldn't otherwise do at the cost of dropping compability of the generated binary? Is that correct? Why did we conclude that it is okay for this to happen? Is it just something C++-pros add that I am not to question further?

Reply:
Maybe I am just allergic to stuff happening in the build system that I do not intuitively grasp...

Reply:
> Googling "-march=native", e.g., yields no result.

Mmh? Did you put in into quotes? Remember that `-term` excludes it.

https://wiki.gentoo.org/wiki/GCC_optimization

That is the first Google result and gives more information that one might fit in a comment.

Reply:
Then just this, maybe?
```
# -march=native see https://wiki.gentoo.org/wiki/GCC_optimization#-march
# -fopenmp-simp for auto vectorization, see [....]
```

Reply:
(Just saying, if I do not know what certain flags do, it's likely other students won't either. Not proud of my lack of googling skills, but if I lack them, again, it's likely others do to.)